### PR TITLE
Feature / Divider

### PR DESCRIPTION
**Pull request by joebell93**
## Description
Base styles and modifier loop to generate `c-divider`.

## Related Issue
https://github.com/sky-uk/toolkit-ui/issues/113

## Motivation and Context
Dividers have been around for a while, but are not yet provided in the Toolkit.

## How Has This Been Tested?
Visual and manual use checks in:
- [x] Chrome
- [x] Firefox
- [x] Safari
- [x] Opera
- [x] IE9

## Screenshots:
screencapture-localhost-9876-1474455873206

IE9
screen shot 2016-09-21 at 16 09 28

## Types of changes
- [x] ~~Bug fix (non-breaking change which fixes an issue)~~
- [x] New feature (non-breaking change which adds functionality)
- [x] ~~Breaking change (fix or feature that would cause existing functionality to change)~~

## Checklist:

- [x] New functions and mixins have appropriate tests.
- [x] All new and existing tests passed.
- [x] Changes have been browser tested (including IE).
- [x] I have added instructions on how to test my changes.
- [x] ~~Package version updated.~~
- [x] CHANGELOG.md updated.

### CODE REVIEW

File: components/_divider.scss [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+ * 1. Set a transparent border on the relevant edge to prevent element collapse.
+ * 2. Reduce the `margin-bottom` by the same width as the divider in order to
+ *    keep on our baseline grid.
+ */
+.c-divider {
+  position: relative;
+  border-width: 0 0 $divider-width 0; /* 1 */
+  border-style: solid;
+  border-color: transparent;
+  margin-bottom: $global-spacing-unit - $divider-width; /* 2 */
```
Comment (MrDinsdale) [inline components/_divider.scss]: Should this not be reset if any direction modifiers are used?

screen shot 2016-09-21 at 12 54 37

i.e. if `c-divider--right` or `c-divider--left` is used should we provide a margin-bottom at all, if so it should be 30px. Presumably the same would be true of `c-divider--top`, especially if set on a div.

**Comment by joebell93** [comment]
@MrDinsdale fixed IE9, just looking into the Safari bug

File: components/_divider.scss (hunk 1 above)
Comment (joebell93) [inline components/_divider.scss]: Yeah potentially, not for top on a `<hr/>` though - so this poses an interesting challenge

**Comment by joebell93** [comment]
@MrDinsdale made some amends, would appreciate another review!

**Comment by joebell93** [comment]
@sky-uk/toolkit any chance I could get some feedback on this please?

**Comment by MrDinsdale** [comment]
Could we get a screen shot of the IE9 fallback?

**Comment by MrDinsdale** [comment]
I think the IE9 fix could do with a little separation from the keyline perhaps but could always look at refining later. Depending on general thoughts about moving the mixins from core I approve.

**Comment by skitson** [comment]
I think the IE9 fix could do with being separate too. If it's the same sort of thing as the one I created then it works fine in IE9 when I tested it.

**Comment by skitson** [comment]
:shipit:

**Pull request merged**